packages/jest-parallel/src/child.js

```diff
@@ -0,0 +1,95 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+'use strict';
+
+import type {JestChildMessage} from './types';
+
+let file = null;
+
+/**
+ * This file is a small bootstrapper for workers. It sets up the communication
+ * between the worker and the parent process, interpreting parent messages and
+ * sending results back.
+ *
+ * The file loaded will be lazily initialized the first time any of the workers
+ * is called. This is done for optimal performance: if the farm is initialized,
+ * but no call is made to it, child Node processes will be consuming the least
+ * possible amount of memory.
+ *
+ * If an invalid message is detected, the child will exit (by throwing) with a
+ * non-zero exit code.
+ */
+process.on('message', (request: JestChildMessage) => {
+  switch (request.type) {
+    case 'initialize':
+      file = request.file;
+      break;
+
+    case 'call':
+      execMethod(request.method, request.args);
+      break;
+
+    case 'end':
+      process.exit(0);
+      break;
+
+    default:
+      throw new TypeError(
+        'Unexpected request from parent process: ' + request.type,
+      );
+  }
+});
+
+function reportSuccess(result: any) {
+  if (!process || !process.send) {
+    throw new Error('Child can only be used when forking a process');
+  }
+
+  process.send({
+    result,
+    type: 'ok',
```

Comment:
instead of a string, could we use a number, like 0 for success and 1 for error? I want to make sure the overhead is as minimal as possible. It should probably be `[0, result]`.